pkgs/by-name/pr/protege/package.nix

```diff
@@ -0,0 +1,104 @@
+{ stdenvNoCC
+, lib
+, fetchFromGitHub
+, copyDesktopItems
+, iconConvTools
+, makeDesktopItem
+, makeWrapper
+, jdk11
+, maven
+}:
+
+let
+  pname = "protege";
+  version = "5.6.4";
+  mvn = maven.override { jdk = jdk11; };
+
+  # Build a platform-independent core from which we can derive platform-specific
+  # directory layouts, so that optimized binary caches can de-duplicate it
+  # effectively.
+  protege-build = mvn.buildMavenPackage {
+    pname = "${pname}-build";
+    inherit version;
+
+    src = fetchFromGitHub {
+      owner = "protegeproject";
+      repo = pname;
+      rev = version;
+      hash = "sha256-Q3MHa7nCeF31n7JPltcemFBc/sJwGA9Ev0ymjQhY/U0=";
+    };
+
+    patches = [
+      ./enforce-plugin-versions.patch
+      ./platform-independent-only.patch
+    ];
+
+    mvnHash = "sha256-GY9qWqRjNQJMd/hoX0S5HIoYtE5pOceZoJRPTkogcao=";
+
+    installPhase = ''
+      runHook preInstall
+
+      mv protege-desktop/target/protege-${version}-platform-independent/Protege-${version} $out
+
+      runHook postInstall
+    '';
+  };
+in
+stdenvNoCC.mkDerivation rec {
```

Comment:
I need it to set the description from the `meta` block as a comment for the desktop file.

What is the benefit of `finalAttrs`? So that a user can override the arguments passed to the derivation and have them reflected by the recursively-inherited attributes? I am not an expert in things that involve fixed-point recursions.
I'll introduce the `finalAttrs` form at the next force-push, as soon as I hear back from you.

Reply:
This is not only about the benefits of finalAttrs, but mainly about the drawbacks of `rec`:

https://nix.dev/guides/best-practices.html#recursive-attribute-set-rec

Reply:
Unfortunately it seems that the `finalAttrs` form is not supported by `buildMavenPackage`. I will use a simple `let ... in`, instead, unless you can suggest a way to obtain the same end result?

Reply:
Yep, finalAttrs is not implemented everywhere yet.